buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.test;
+
+import java.nio.file.Path;
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.Queue;
+import java.util.LinkedList;
+import java.lang.StringBuilder;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.LinkedHashMap;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Assertions;
+import org.apache.calcite.buildtools.parser.DialectGenerate;
+import org.apache.calcite.buildtools.parser.DialectGenerate.CurlyParser;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DialectGenerateTest {
+
+  /**
+   * Retrns a DialectGenerate with root path of calcite/parsingTest and
+   * dialect path of calcite/parsingTest/intermediate/testDialect.
+   */
+  private DialectGenerate setupDialectGenerate() {
+    Path rootPath = Paths.get("..", "..", "..", "parsingTest");
+    // Adds the path /intermediate/testDialect/ to the end of rootPath.
+    Path dialectPath = rootPath.resolve(Paths.get("intermediate", "testDialect"));
+    File rootFile = rootPath.toFile();
+    File dialectFile = dialectPath.toFile();
+    return new DialectGenerate(dialectFile, rootFile, "");
+  }
+
+  /**
+   * Returns the expected extraction results after calling DialectGenerate.extractFunctions
+   * onto the parsingTest directory structure.
+   */
+  private Map<String, String> getExpectedExtractionResults() {
```

Comment:
The format of these test case strings is difficult to grok. And it could be hard for a future developer to add new test cases to.

Suggest putting the raw text in a separate test data file that this test reads from.

Reply:
I tried making it as simple as possible by having a new line where the function would have a new line. I'd prefer not having to deal with additional file IO but if you deem it necessary I'll try to change it. Any suggestion on the format of the test file?

Reply:
Even if it takes a little extra work, it's good practice to follow common patterns. The pattern we should be following here is using a golden file. Here's some information about it:

https://softwareengineering.stackexchange.com/questions/358786/what-is-golden-files